catalyst/experiments/hydra_config.py

```diff
@@ -0,0 +1,552 @@
+    """
+    Experiment created from a hydra configuration file.
+    """
+
+    def __init__(self, cfg: DictConfig):
+        """
+        Args:
+            cfg (dict): dictionary with parameters
+        """
+        self._config: DictConfig = deepcopy(cfg)
+        self._trial = None
+        self._initial_seed: int = self._config.args.seed
+        self._verbose: bool = self._config.args.verbose
+        self._check_time: bool = self._config.args.timeit
+        self._check_run: bool = self._config.args.check
+        self._overfit: bool = self._config.args.overfit
+        self._logdir: str = os.getcwd()
+
+    @property
+    def initial_seed(self) -> int:
+        """Experiment's initial seed value."""
+        return self._initial_seed
+
+    @property
+    def logdir(self) -> str:
+        """Path to the directory where the experiment logs."""
+        return self._logdir
+
+    @property
+    def hparams(self) -> OrderedDict:
+        """Returns hyperparameters"""
+        return OrderedDict(OmegaConf.to_container(self._config, resolve=True))
+
+    @property
+    def trial(self) -> Any:
+        """Returns hyperparameter trial for current experiment"""
+        return self._trial
+
+    @property
+    def distributed_params(self) -> Dict:
+        """Dict with the parameters for distributed and FP16 methond."""
+        return self._config.distributed
+
+    @property
+    def stages(self) -> List[str]:
+        """Experiment's stage names."""
+        stages_keys = list(self._config.stages.keys())
+        return stages_keys
+
+    def get_stage_params(self, stage: str) -> Mapping[str, Any]:
+        """Returns the state parameters for a given stage."""
+        return self._config.stages[stage].params
+
+    @staticmethod
+    def _get_model(params: DictConfig) -> Model:
+        model: Model = hydra.utils.instantiate(params)
+        return model
+
+    def get_model(self, stage: str) -> Dict[str, Model]:
+        """Returns the models for a given stage."""
+        assert "models" in self._config, "config must contain 'models' key"
+        models_params: DictConfig = self._config.models
+        model: Dict[str, Model] = {
+            name: self._get_model(model_params)
+            for name, model_params in models_params.items()
+        }
+        return model
+
+    @staticmethod
+    def _get_criterion(params: DictConfig) -> Criterion:
+        criterion: Criterion = hydra.utils.instantiate(params)
+        return criterion
+
+    def get_criterion(self, stage: str) -> Dict[str, Criterion]:
+        """Returns the criterions for a given stage."""
+        if "criterions" not in self._config.stages[stage]:
+            return {}
+        criterions_params: DictConfig = self._config.stages[stage].criterions
+        criterion: Dict[str, Criterion] = {
+            name: self._get_criterion(criterion_params)
+            for name, criterion_params in criterions_params.items()
+        }
```

Comment:
is there any support for non-key-value criterion?

Reply:
No. See previous answer.